scalafix-core/shared/src/main/scala/scalafix/patch/Patch.scala

```diff
@@ -205,7 +205,7 @@ object Patch {
       case _: LintPatch => hasLintMessage = true
       case _ => onlyLint = false
     }
-    hasLintMessage && onlyLint
+    patch.isEmpty || hasLintMessage && onlyLint
```

Comment:
I wrote an empty syntactic rule and I ran the SemanticSuite. It complained it was missing an output. This is because it checks if there are lint messages. It's obviously not the best way to fix this.